src/main/java/com/powsybl/sld/server/NetworkAreaDiagramService.java

```diff
@@ -38,26 +38,57 @@ class NetworkAreaDiagramService {
     @Autowired
     private NetworkStoreService networkStoreService;
 
-    public Pair<String, Integer> generateNetworkAreaDiagramSvg(UUID networkUuid, String variantId, List<String> voltageLevelsIds, int depth) {
+    public SvgAndMetadata generateNetworkAreaDiagramSvg(UUID networkUuid, String variantId, List<String> voltageLevelsIds, int depth) {
         Network network = DiagramUtils.getNetwork(networkUuid, variantId, networkStoreService, PreloadingStrategy.COLLECTION);
         voltageLevelsIds.forEach(voltageLevelId -> {
             if (network.getVoltageLevel(voltageLevelId) == null) {
                 throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Voltage level" + voltageLevelId + " not found");
             }
         });
 
-        VoltageLevelFilter vlFilter = VoltageLevelFilter.createVoltageLevelsDepthFilter(network, voltageLevelsIds, depth);
-
         try (StringWriter svgWriter = new StringWriter()) {
             SvgParameters svgParameters = new SvgParameters()
                     .setSvgWidthAndHeightAdded(true)
                     .setCssLocation(SvgParameters.CssLocation.EXTERNAL_NO_IMPORT);
             LayoutParameters layoutParameters = new LayoutParameters();
             StyleProvider styleProvider = new TopologicalStyleProvider(network);
-            new NetworkAreaDiagram(network, voltageLevelsIds, depth).draw(svgWriter, svgParameters, layoutParameters, styleProvider);
-            return Pair.of(svgWriter.toString(), vlFilter.getNbVoltageLevels());
+            new NetworkAreaDiagram(network, voltageLevelsIds, depth)
+                    .draw(svgWriter, svgParameters, layoutParameters, styleProvider);
+
+            Map<String, Object> additionalMetadata = computeAdditionalMetadata(network, voltageLevelsIds, depth);
+
+            return SvgAndMetadata.builder()
+                    .svg(svgWriter.toString())
+                    .additionalMetadata(additionalMetadata).build();
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
+
+    private Map<String, Object> computeAdditionalMetadata(Network network, List<String> voltageLevelsIds, int depth) {
+
+        VoltageLevelFilter vlFilter = VoltageLevelFilter.createVoltageLevelsDepthFilter(network, voltageLevelsIds, depth);
+
+        // list of {"id": id, "name": name, "substationId": substationId}
+        List<Map<String, String>> voltageLevelsInfos = voltageLevelsIds.stream()
```

Comment:
Add static constants for metadata keys (voltageId, voltageName, subtsationId) in SvgAndMetadata class
So all matadata names are defined centrally
Perhaps a dto **VoltageLevelInfos**

Reply:
At first we said we should do like that but I agree it's probably better if we use DTO instead of a map of String.
I'm doing it.